honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,330 @@
+
+
+def mesh_to_mesh3d(mesh, color_by_face=True):
+    """Get a Ladybug Mesh3D object from a Rhino3dm Mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
+    """Get a list of Ladybug Face3D objects from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+    return faces
+
+
+def check_planarity(brep):
+    """Check the planarity of a Brep.
+
+    Args:
+        brep: Geometry of a Rhino3dm Brep
+
+    Returns:
+        Bool. True if planar otherwise False
+    """
+    is_planar = [brep.Faces[i].UnderlyingSurface().IsPlanar()
+        for i in range(len(brep.Faces))]
+    return all(is_planar)
+
+
+def brep2d_to_face3d(brep, tolerance):
+    """Get a list of Ladybug Face3D objects from a planar rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+        tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # * Check 01 - If any of the edge is an arc
+    lines = []
+
+    # Getting all vertices of the face
+    mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
+    mesh_pts = [to_point3d(mesh.Vertices[i])
+                for i in range(len(mesh.Vertices))]
+
+    curved = False
+    for i in range(len(brep.Edges)):
+        if not brep.Edges[i].IsLinear(tolerance):
+            curved = True
+            break
+    # If one of the edges is curved, mesh it
+    if curved:
+        warnings.warn(
+            'A Brep has curved edges. It will be meshed.')
+        faces.extend(brep_to_face3d(brep))
+        return faces
+    else:
+        # Create Ladybug lines from start and end points of edges
+        for i in range(len(brep.Edges)):
+            start_pt = to_point3d(brep.Edges[i].PointAtStart)
+            end_pt = to_point3d(brep.Edges[i].PointAtEnd)
+            line = LineSegment3D.from_end_points(start_pt, end_pt)
+            lines.append(line)
+
+        # Create Ladybug Polylines from the lines
+        polylines = Polyline3D.join_segments(lines, tolerance)
+
+        # In the list of the Polylines, if there's only one polyline then
+        # the face has no holes
+        if len(polylines) == 1:
+            if len(mesh_pts) == 4 or len(mesh_pts) == 3:
+                faces.extend(mesh_to_face3d(mesh))
```

Comment:
You should move this check all of the way to the beginning of the `else:` above.  If there are only 3 or 4 vertices, then there's no need to create a join the edges at all.  Not creating the edge objects is going to save a lot of time and memory in the case that the Rhino model has a lot of rectangular surfaces.

Reply:
Complied.